pkg/cmd/server/kubernetes/node/node.go

```diff
@@ -476,8 +476,6 @@ func (c *NodeConfig) RunProxy() {
 	endpointsConfig.RegisterEventHandler(endpointsHandler)
 	go endpointsConfig.Run(utilwait.NeverStop)
 
-	recorder.Eventf(c.ProxyConfig.NodeRef, kapi.EventTypeNormal, "Starting", "Starting kube-proxy.")
```

Comment:
It doesn't look like anything would have ever modified c.ProxyConfig.NodeRef after it was created in the old code. So you could just move its creation to here and do

    nodeRef := &kclientv1.ObjectReference{
            Kind:      "Node",
            Name:      hostname,
            UID:       ktypes.UID(hostname),
            Namespace: "",
    }
    recorder.Eventf(nodeRef, kapi.EventTypeNormal, "Starting", "Starting kube-proxy.")

(based on the current NodeRef definition in kubernetes/cmd/kube-proxy/app/server.go)